### PR TITLE
eos issue#1743 bug fix get_bgp_neighbors()

### PR DESCRIPTION
**Pull request by vaderisback77**
Testing using updated NEIGHBOR_FILTER value

--snipped output for IPv4 and IPv6 BGP neighbors--

```

## BEFORE

>>> print(pprint(output_neighbor_cmds))
[{'
            'BGP neighbor is 110.1.1.1, remote AS 60001, external link\n'
            '  BGP version 4, remote router ID 194.0.0.1, VRF VPN-DRG-2555\n'
            '  BGP state is Idle, Could not find interface for peer\n'
            '    IPv4 Unicast:                     0         0              '
            '0                   0\n'
            '    IPv6 Unicast:                     0         0              '
            '0                   0\n'
            'Local AS is 60000, local router ID 12.12.12.12\n'
            'BGP neighbor is 111.1.1.1, remote AS 31898, external link\n'
            '  BGP version 4, remote router ID 111.1.1.1, VRF VPN-DRG-2555\n'
            '  BGP state is Established, up for 1d04h\n'
            '    IPv4 Unicast:                     1         0              '
            '0                   0\n'
            '    IPv6 Unicast:                     0         0              '
            '0                   0\n'
            'Local AS is 60000, local router ID 12.12.12.12\n'
            '  IPv4 Unicast: 111.1.1.0\n'

             'BGP neighbor is 2607:9b80:8000:187::5, remote AS 31898, external '
            'link\n'
            '  BGP version 4, remote router ID 111.1.27.1, VRF VPN-DRG-2581\n'
            '  BGP state is Established, up for 1d04h\n'
            '    IPv4 Unicast:                     0         0              '
            '0                   0\n'
            '    IPv6 Unicast:                     0         0              '
            '0                   0\n'
            'Local AS is 60000, local router ID 111.1.27.0\n'
            '  IPv6 Unicast: 2607:9b80:8000:187::4\n'


## AFTER

>>> pprint(output_neighbor_cmds)
[{'output': 'BGP neighbor is 110.1.1.1, remote AS 60001, external link\n'
            '  BGP version 4, remote router ID 0.0.0.0, VRF VPN-DRG-2555\n'
            '  BGP state is Idle, Could not find interface for peer\n'
            '    IPv4 Unicast:                     0         0              '
            '0                   0\n'
            '    IPv6 Unicast:                     0         0              '
            '0                   0\n'
            'Local AS is 60000, local router ID 12.12.12.12\n'
            'BGP neighbor is 111.1.1.1, remote AS 31898, external link\n'
            '  BGP version 4, remote router ID 111.1.1.1, VRF VPN-DRG-2555\n'
            '  BGP state is Established, up for 00:33:24\n'
            '    IPv4 Unicast:                     1         0              '
            '0                   0\n'
            '    IPv6 Unicast:                     0         0              '
            '0                   0\n'
            'Local AS is 60000, local router ID 12.12.12.12\n'

            'BGP neighbor is 2607:9b80:8000:187::5, remote AS 31898, external '
            'link\n'
            '  BGP version 4, remote router ID 111.1.27.1, VRF VPN-DRG-2581\n'
            '  BGP state is Established, up for 00:33:13\n'
            '    IPv4 Unicast:                     0         0              '
            '0                   0\n'
            '    IPv6 Unicast:                     0         0              '
            '0                   0\n'
            'Local AS is 60000, local router ID 111.1.27.0\n'

```

### CODE REVIEW

**Comment by vaderisback77** [comment]
The PR is failing the **build / std_tests (3.9) (pull_request)** and I am trying to get some details on this failure and what needs to be done to fix this (Is this on my end?). The tests output is not super helpful as where exactly its failing. I would appreciate any input on this so we are able to get this merge soon. TIA

```
test/eos/test_getters.py::TestGetter::test_get_bgp_neighbors[issue1168] FAILED
test/eos/test_getters.py::TestGetter::test_get_bgp_neighbors[issue944] FAILED
test/eos/test_getters.py::TestGetter::test_get_bgp_neighbors[issue1356] FAILED
test/eos/test_getters.py::TestGetter::test_get_bgp_neighbors[issue58_neighbor_down] FAILED
test/eos/test_getters.py::TestGetter::test_get_bgp_neighbors[normal] FAILED
```

when I ran pytest locally against py3.9, I am running into module missing errors 

```
(napalm-3.9) ➜  napalm git:(vaderisback77/issue-1743/bug-fix/get-bgp-neighbors-7050cx3m) pip freeze | grep napalm
-e git+https://github.com/vaderisback77/napalm.git@8ec78e93b9e1f8c2d7fb2473d6e7befbd7b33eee#egg=napalm
(napalm-3.9) ➜  napalm git:(vaderisback77/issue-1743/bug-fix/get-bgp-neighbors-7050cx3m) 
(napalm-3.9) ➜  napalm git:(vaderisback77/issue-1743/bug-fix/get-bgp-neighbors-7050cx3m) python -V
Python 3.9.5 
(napalm-3.9) ➜  napalm git:(vaderisback77/issue-1743/bug-fix/get-bgp-neighbors-7050cx3m) pytest test 
================================================================================================================================================================================ test session starts =================================================================================================================================================================================
platform darwin -- Python 3.9.5, pytest-7.1.3, pluggy-1.0.0
rootdir: /Users/saurasar/Documents/ocina/git/napalm, configfile: setup.cfg
collected 463 items / 4 errors                                                                                                                                                                                                                                                                                                                                                       

======================================================================================================================================================================================= ERRORS =======================================================================================================================================================================================
_______________________________________________________________________________________________________________________________________________________________ ERROR collecting test/base/test_get_network_driver.py ________________________________________________________________________________________________________________________________________________________________
ImportError while importing test module '/Users/saurasar/Documents/ocina/git/napalm/test/base/test_get_network_driver.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test/base/test_get_network_driver.py:3: in <module>
    from ddt import ddt, data
E   ModuleNotFoundError: No module named 'ddt'
_____________________________________________________________________________________________________________________________________________________________________ ERROR collecting test/eos/test_heredoc.py ______________________________________________________________________________________________________________________________________________________________________
ImportError while importing test module '/Users/saurasar/Documents/ocina/git/napalm/test/eos/test_heredoc.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test/eos/test_heredoc.py:1: in <module>
    import mock
E   ModuleNotFoundError: No module named 'mock'
_____________________________________________________________________________________________________________________________________________________________________ ERROR collecting test/nxos/test_getters.py _____________________________________________________________________________________________________________________________________________________________________
ImportError while importing test module '/Users/saurasar/Documents/ocina/git/napalm/test/nxos/test_getters.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test/nxos/test_getters.py:8: in <module>
    from mock import patch
E   ModuleNotFoundError: No module named 'mock'
___________________________________________________________________________________________________________________________________________________________________ ERROR collecting test/nxos_ssh/test_getters.py ___________________________________________________________________________________________________________________________________________________________________
ImportError while importing test module '/Users/saurasar/Documents/ocina/git/napalm/test/nxos_ssh/test_getters.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test/nxos_ssh/test_getters.py:8: in <module>
    from mock import patch
E   ModuleNotFoundError: No module named 'mock'
================================================================================================================================================================================== warnings summary ==================================================================================================================================================================================
../../../../.virtualenvs/napalm-3.9/lib/python3.9/site-packages/_pytest/config/__init__.py:1253
  /Users/saurasar/.virtualenvs/napalm-3.9/lib/python3.9/site-packages/_pytest/config/__init__.py:1253: PytestConfigWarning: Unknown config option: json_report
  
    self._warn_or_fail_if_strict(f"Unknown config option: {key}\n")

../../../../.virtualenvs/napalm-3.9/lib/python3.9/site-packages/_pytest/config/__init__.py:1253
  /Users/saurasar/.virtualenvs/napalm-3.9/lib/python3.9/site-packages/_pytest/config/__init__.py:1253: PytestConfigWarning: Unknown config option: jsonapi
  
    self._warn_or_fail_if_strict(f"Unknown config option: {key}\n")

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================================================================================================================================================================== short test summary info ===============================================================================================================================================================================
ERROR test/base/test_get_network_driver.py
ERROR test/eos/test_heredoc.py
ERROR test/nxos/test_getters.py
ERROR test/nxos_ssh/test_getters.py
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 4 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
=========================================================================================================================================================================== 2 warnings, 4 errors in 2.67s ============================================================================================================================================================================

```

**Comment by vaderisback77** [comment]
Local pytest checks passing after adding missing mocked data files and dependencies

```
(napalm-3.9) ➜  napalm git:(vaderisback77/issue-1743/bug-fix/get-bgp-neighbors-7050cx3m) ✗ pytest test
============================================================================================ test session starts ============================================================================================
platform darwin -- Python 3.9.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /Users/saurasar/Documents/ocina/git/napalm, configfile: setup.cfg
plugins: pylama-8.2.1, json-0.4.0, cov-3.0.0
collected 591 items                                                                                                                                                                                         

test/base/test_get_network_driver.py .............                                                                                                                                                    [  2%]
test/base/test_helpers.py .............                                                                                                                                                               [  4%]
test/base/test_mock_driver.py ..........                                                                                                                                                              [  6%]
test/base/test_napalm_test_framework.py sssssssssssssssssssssssssssssss                                                                                                                               [ 11%]
test/base/validate/test_unit.py ............................                                                                                                                                          [ 16%]
test/base/validate/test_validate.py ............                                                                                                                                                      [ 18%]
test/eos/test_cli_syntax.py ....                                                                                                                                                                      [ 18%]
test/eos/test_getters.py ...............................ss...........ss.................s.                                                                                                            [ 29%]
test/eos/test_heredoc.py ...                                                                                                                                                                          [ 30%]
test/eos/test_versions.py ..                                                                                                                                                                          [ 30%]
test/ios/test_getters.py .....................................................................s..s...............s..                                                                                  [ 46%]
test/iosxr/test_getters.py ................ss......s...s..s...sss                                                                                                                                     [ 52%]
test/iosxr_netconf/test_getters.py ......................ss......s...s..s.sssss                                                                                                                       [ 59%]
test/junos/test_getters.py ...........................s..........s..............s...                                                                                                                  [ 69%]
test/nxapi_plumbing/test_build_payload.py .....                                                                                                                                                       [ 70%]
test/nxapi_plumbing/test_config.py .......                                                                                                                                                            [ 71%]
test/nxapi_plumbing/test_import.py .                                                                                                                                                                  [ 71%]
test/nxapi_plumbing/test_show_jsonrpc.py ......                                                                                                                                                       [ 72%]
test/nxapi_plumbing/test_show_xml.py .....                                                                                                                                                            [ 73%]
test/nxos/test_getters.py .....s.......ss...ss........ss.ss.....s....s.....                                                                                                                           [ 81%]
test/nxos_ssh/test_getters.py .................ss....s...s.........s.ss........ss............                                                                                                         [ 92%]
test/pyiosxr/test_iosxr.py ............................................                                                                                                                               [100%]

============================================================================================= warnings summary ==============================================================================================
test/iosxr_netconf/test_getters.py::TestGetter::test_get_config[normal]
  /Users/saurasar/Documents/ocina/git/napalm/napalm/iosxr_netconf/iosxr_netconf.py:3142: FutureWarning: The behavior of this method will change in future versions. Use specific 'len(elem)' or 'elem is not None' test instead.
    if cli_tree:

-- Docs: https://docs.pytest.org/en/stable/warnings.html
--------------------------------------------------------------- generated json report: /Users/saurasar/Documents/ocina/git/napalm/report.json ---------------------------------------------------------------
================================================================================ 511 passed, 80 skipped, 1 warning in 9.44s =================================================================================
(napalm-3.9) ➜  napalm git:(vaderisback77/issue-1743/bug-fix/get-bgp-neighbors-7050cx3m) ✗ 
```

File: napalm/eos/eos.py [hunk 1]
```diff
@@ -664,7 +664,7 @@ def get_re_group(res, key, default=None):
             except KeyError:
                 return default
 
-        NEIGHBOR_FILTER = "bgp neighbors vrf all | include remote AS | remote router ID |IPv[46] (Unicast|6PE):.*[0-9]+|^Local AS|Desc|BGP state"  # noqa
+        NEIGHBOR_FILTER = "bgp neighbors vrf all | include IPv[46] (Unicast|6PE):.*[0-9]+ | grep -v ' IPv[46] Unicast:/.' | remote AS |^Local AS|Desc|BGP state |remote router ID"  # noqa
```
Comment (bewing) [inline napalm/eos/eos.py:667]: This regex is starting to get pretty complex.  Is it time to refactor this to TextFSM?

**Review by bewing — COMMENTED** [review]
Do we know if this is JUST platform-related, or did the output update in other devices running the latest EOS versions?

**Comment by vaderisback77** [comment]
> Do we know if this is JUST platform-related, or did the output update in other devices running the latest EOS versions?

We noticed this issue during the onboarding of 7050CX3M platform with 4.27.2F-DPE, I am running same tests against few other platforms we have in the lab, and will get back with my findings.

**Comment by vaderisback77** [comment]
> > Do we know if this is JUST platform-related, or did the output update in other devices running the latest EOS versions?
> 
> We noticed this issue during the onboarding of 7050CX3M platform with 4.27.2F-DPE, I am running same tests against few other platforms we have in the lab, and will get back with my findings. I didn't see this same issue when i earlier tested on 7050TX running 4.24.x

**Comment by vaderisback77** [comment]
> Do we know if this is JUST platform-related, or did the output update in other devices running the latest EOS versions?

I tested in another platform, DCS-7050SX3-48YC8-F with the same code version(4.27.2F-DPE), and I am able to reproduce the issue. This extra line ('  IPv[46] Unicast: 111.1.1.0\n') is getting added for established BGP neighbors


```
            '  BGP version 4, remote router ID 111.1.1.1, VRF VPN-DRG-2555\n'
            '  BGP state is Established, up for 10d09h\n'
            '    IPv4 Unicast:                     1         0              '
            '0                   0\n'
            '    IPv6 Unicast:                     0         0              '
            '0                   0\n'
            'Local AS is 60000, local router ID 12.12.12.12\n'
            '  IPv4 Unicast: 111.1.1.0\n'
         
 >>> k.get_bgp_neighbors()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/saurasar/Documents/ocina/git/napalm/napalm/eos/eos.py", line 750, in get_bgp_neighbors
    "remote_as": napalm.base.helpers.as_number(neighbor_info.group("as")),
AttributeError: 'NoneType' object has no attribute 'group'

```

File: napalm/eos/eos.py (hunk 1 above)
Comment (vaderisback77) [reply napalm/eos/eos.py:667]: > This regex is starting to get pretty complex. Is it time to refactor this to TextFSM?

Are you suggesting me to add TextFSM?

Comment (bewing) [reply napalm/eos/eos.py:667]: Not really, just thinking aloud, for things to spend my copious spare time on.

**Comment by bewing** [comment]
I did some local testing today, and this change overall looks good and doesn't appear to cause any regressions.  However, can you please `git rm` the old/unused `show_ip_bgp_neighbors_vrf_all__include_remote...` files from the test cases, as they will no longer be used after this change?

**Comment by vaderisback77** [comment]
> I did some local testing today, and this change overall looks good and doesn't appear to cause any regressions. However, can you please `git rm` the old/unused `show_ip_bgp_neighbors_vrf_all__include_remote...` files from the test cases, as they will no longer be used after this change?

done.

**Review by bewing — APPROVED** [review]
LGTM

**Pull request merged**